### DISCUSSION TITLE
Inefficient object listing loading (N+1 problem)

### DISCUSSION THREAD

**Discussion by root913** [pimcore/pimcore#15742] [Q&A / Support]
I have notices that in working with Pimcore that listing of object generates many database queries that the only difference between them is object id. 
**Here is listing of 20 objects in grid:**
![image](https://github.com/pimcore/pimcore/assets/3902450/184ee4fa-fb68-4978-884b-a4e7def2211e)

**This is caused by this fragment:**
https://github.dev/pimcore/pimcore/blob/727ab1a73020f540ac1d2c12386292909e6b3125/models/DataObject/Listing/Dao.php#L66-L68
![image](https://github.com/pimcore/pimcore/assets/3902450/3f9c7069-ad16-4e53-8b5a-ac8dfddd6d6e)
In this fragment we are executing getById query for every object id. 
This could be changed to something like "WHERE o_id IN (1, 2, 3, 4)...".

Is there any reason this is made like this? This is classic N+1 problem. 
I'm starting this discussion to get new perspective and maybe some answers.

**Comment by dpfaffenbauer** [comment]
I'l like the idea, maybe you can PR it?

**Reply by root913** [reply]
Yes, I have some prototype. Maybe next week I'll create PR.
